src/screens/HyperSwitch/SwitchMerchant/SwitchMerchant.res

```diff
@@ -146,41 +123,36 @@ module ExternalUser = {
   @react.component
   let make = (~switchMerchant, ~isAddMerchantEnabled) => {
     open UIUtils
-    open APIUtils
-    let fetchDetails = useGetMethod()
     let defaultMerchantId = HSLocalStorage.getFromMerchantDetails("merchant_id")
+    let switchMerchantList = Recoil.useRecoilValueFromAtom(HyperswitchAtom.switchMerchantListAtom)
     let merchantDetailsTypedValue =
       HSwitchUtils.useMerchantDetailsValue()->MerchantAccountUtils.getMerchantDetails
     let (selectedMerchantObject, setSelectedMerchantObject) = React.useState(_ => {
       merchant_id: defaultMerchantId,
       merchant_name: defaultMerchantId,
+      is_active: false,
     })
     let (showModal, setShowModal) = React.useState(_ => false)
     let (options, setOptions) = React.useState(_ => [])
 
-    let fetchMerchantIDs = async () => {
-      let url = getURL(~entityName=USERS, ~userType=#SWITCH_MERCHANT, ~methodType=Get, ())
-      try {
-        let res = await fetchDetails(url)
-        let typedValueOfResponse = res->convertListResponseToTypedResponse
-        setOptions(_ => typedValueOfResponse)
-        let extractMerchantObject =
-          typedValueOfResponse
-          ->Array.find(ele => ele.merchant_id === defaultMerchantId)
-          ->Option.getOr({
-            merchant_id: defaultMerchantId,
-            merchant_name: defaultMerchantId,
-          })
-        setSelectedMerchantObject(_ => extractMerchantObject)
-      } catch {
-      | _ => ()
-      }
+    let fetchMerchantIDs = () => {
+      let filteredSwitchMerchantList = switchMerchantList->Array.filter(ele => ele.is_active)
+      setOptions(_ => filteredSwitchMerchantList)
+      let extractMerchantObject =
+        switchMerchantList
+        ->Array.find(ele => ele.merchant_id === defaultMerchantId)
+        ->Option.getOr({
+          merchant_id: defaultMerchantId,
+          merchant_name: defaultMerchantId,
+          is_active: false,
+        })
+      setSelectedMerchantObject(_ => extractMerchantObject)
     }
 
-    React.useEffect1(() => {
-      fetchMerchantIDs()->ignore
+    React.useEffect2(() => {
+      fetchMerchantIDs()
       None
-    }, [merchantDetailsTypedValue.merchant_name])
+    }, (merchantDetailsTypedValue.merchant_name, switchMerchantList))
```

Comment:
please check the length of `switchMerchantList` in the useEffect

Reply:
Updated